### DISCUSSION TITLE
Import Snippet Section from XML file

### DISCUSSION THREAD

**Discussion by maciejwalkowiak** [facelessuser/pymdown-extensions#1914] [Q&A] [answered]
I want to use Snippet Sections in an XML file:

```xml
<dependencies>
    <!-- --8<-- [start:func] -->
    <dependency>
        <groupId>software.amazon.awssdk</groupId>
        <artifactId>dynamodb-enhanced</artifactId>
    </dependency>
    <!-- --8<-- [end:func] -->
</dependencies>
```

`--8<--` within an XML comment makes the XML invalid which as a result breaks the XML tooling. The error is following:

```
in comment after two dashes (--) next character must be > not 8 (position: START_TAG seen ...<dependencies>\n        <!-- --8... @21:17) 
```

Is there a way to use different "tag" than `--8<--` or any other way to make snippet sections work with XML files?

**Comment by facelessuser** [comment]
Ugh, XML...I completely forgot about this annoying rule in XML where you cannot have `--` in a comment. I'll never understand why they couldn't just parse whatever is between the start and the end delimiter. There are other ways to prevent nested comments, like just not treating them as nested comments. I digress.

I don't exactly recall why I required at least 2 `-`, I imagine it was quite arbitrary, but I see no reason we cannot allow just one. I think what we can do is just relax the syntax a little more. `-8<-` should still be plenty unique, will work with XML, and will be backward compatible.

I think doing the above should be more than sufficient. The idea was to allow these to live in comments. I'm fairly certain that fixing this for XML will allow this to work pretty much everywhere. If we run into this issue again in some other language, we can explore alternate syntax as a last resort.

**Reply by facelessuser** [reply]
Bug opened to resolve this: https://github.com/facelessuser/pymdown-extensions/issues/1916

**Comment by facelessuser** [comment, marked as answer]
9.9.2 resolves this issue by allowing `-8<-` (single dash) moving forward. This should be compatible with XML Marking this answered.

**Reply by maciejwalkowiak** [reply]
Thanks!
